powerplantmatching/osm/quality/coverage.py

```diff
@@ -0,0 +1,881 @@
+    countries_to_check : list[str], optional
+        Specific country codes to check live counts for
+    show_outdated_only : bool
+        Only show countries with outdated data
+    outdated_threshold : float
+        Ratio below which cache is considered outdated
+
+    Returns
+    -------
+    dict or None
+        If return_data=True, returns coverage statistics dict.
+        Otherwise prints report and returns None.
+
+    Examples
+    --------
+    >>> # Print coverage report
+    >>> show_country_coverage(show_missing=True)
+
+    >>> # Get data for analysis
+    >>> data = show_country_coverage(return_data=True)
+    >>> print(f"Countries cached: {data['countries_cached']}")
+
+    >>> # Check if Germany needs update
+    >>> show_country_coverage(
+    ...     check_live_counts=True,
+    ...     countries_to_check=['DE']
+    ... )
+
+    Notes
+    -----
+    Live count checking makes API calls and can be slow for many countries.
+    The coverage report includes breakdowns by continent and cache status.
+    """
+    config = get_config()
+    osm_config = config.get("OSM", {})
+
+    if cache_dir is None:
+        cache_dir, _ = get_osm_cache_paths(config)
+
+    cache = ElementCache(cache_dir)
+
+    if not return_data:
+        print("Loading cache data...")
+
+    cache.load_all_caches()
+
+    all_countries = {c.alpha_2: c.name for c in pycountry.countries}  # type: ignore
+    total_possible_countries = len(all_countries)
+
+    cached_countries = {}
+    total_plants = 0
+    total_generators = 0
+
+    for country_code, data in cache.plants_cache.items():
+        if country_code not in cached_countries:
+            cached_countries[country_code] = {
+                "name": all_countries.get(country_code, f"Unknown ({country_code})"),
+                "plants_cached": 0,
+                "generators_cached": 0,
+                "total_cached": 0,
+                "plants_live": None,
+                "generators_live": None,
+                "total_live": None,
+                "plants_diff": None,
+                "generators_diff": None,
+                "cache_status": "unknown",
+            }
+        plants_count = len(data.get("elements", []))
+        cached_countries[country_code]["plants_cached"] = plants_count
+        total_plants += plants_count
+
+    for country_code, data in cache.generators_cache.items():
+        if country_code not in cached_countries:
+            cached_countries[country_code] = {
+                "name": all_countries.get(country_code, f"Unknown ({country_code})"),
+                "plants_cached": 0,
+                "generators_cached": 0,
+                "total_cached": 0,
+                "plants_live": None,
+                "generators_live": None,
+                "total_live": None,
+                "plants_diff": None,
+                "generators_diff": None,
+                "cache_status": "unknown",
+            }
```

Comment:
I would rather iterate over the country_keys once for initialization:

```
for country_code in (cache.plants_cache.keys() | cache.generators_cache.keys()):
   cached_countries[country_code] = ...
```

This reduces duplicated code